cmd/kube-controller-manager/controller-manager.go

```diff
@@ -46,10 +51,35 @@ func main() {
 	util.InitLogs()
 	defer util.FlushLogs()
 
-	verflag.PrintAndExitIfRequested()
+	//Functions to start and stop this daemon.
+	startRC := func(leaseUserInfo *ha.LeaseUser) bool {
+		leaseUserInfo.Running = true
+		if err := s.Run(pflag.CommandLine.Args()); err != nil {
+			fmt.Fprintf(os.Stderr, "%v\n", err)
+			glog.Infof("EXITING NOW ! Killed")
+			os.Exit(1)
+		}
+		return true
+	}
+
+	endRC := func(leaseUserInfo *ha.LeaseUser) bool {
+		glog.Infof("Hard-exiting the replication controller process now !")
+
+		//Even though we're exiting, we should set this flag before dying just for completeness.
+		leaseUserInfo.Running = false
+		os.Exit(0)
+
+		//this should never be reached
+		return true
+	}
+
+	//This starts a thread that continues running.
+	ha.RunHA(s.Kubeconfig, s.Master, startRC, endRC, "ha.cm.lock")
 
-	if err := s.Run(pflag.CommandLine.Args()); err != nil {
-		fmt.Fprintf(os.Stderr, "%v\n", err)
-		os.Exit(1)
+	for true {
+		glog.Infof("CM lease loop is running...")
+		time.Sleep(5 * time.Second)
```

Comment:
i added a 5 second default.  That way even if we dont set reasonable defaults in the top level, it will still poll the lease using a reasonable interval.